sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -598,6 +601,46 @@ class FilesElement extends Polymer.Element {
     }
   }
 
+  _toggleAltAddToolbar() {
+    this.$.altAddToolbar.toggle();
+  }
+
+  _toggleAltUpdateToolbar() {
+    this.$.altUpdateToolbar.toggle();
+  }
+
+  /**
+   * Called on window.resize, collapses elements to keep the element usable
+   * on small screens.
+   */
+  _resizeHandler() {
+    const width = this.$.toolbar.clientWidth;
+    // Collapse the add buttons on the toolbar
+    if (width < this._addToolbarCollapseThreshold) {
+      Utils.moveElementChildren(this.$.addToolbar, this.$.altAddToolbar);
+      this.$.altAddToolbarToggle.style.display = "inline-flex";
+    } else {
+      Utils.moveElementChildren(this.$.altAddToolbar, this.$.addToolbar);
+      this.$.altAddToolbarToggle.style.display = "none";
+      this.$.altAddToolbar.close();
+    }
+
+    // Collapse the file operations buttons on the toolbar
+    if (width < this._updateToolbarCollapseThreshold) {
```

Comment:
I think creating a function could be clear enough if you could come up with a good descriptive name for the it, something like collapseOrUncollapseToolbarSegment. As it is, I think some of your comments are perhaps a bit confusing, they talk about collapsing elements, but this code also uncollapses elements if the size is large. But refactoring would not reduce the code by that much, it's more for preventing code drift in the future, where someone might add something to one and not the other, so I'm OK with your leaving it as is.